internal/metrics/submitter/main.go

```diff
@@ -0,0 +1,103 @@
+package submitter
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/stackrox/image-prefetcher/internal/metrics/gen"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+type Submitter struct {
+	c      chan *gen.Result
+	done   chan struct{}
+	client gen.MetricsClient
+	logger *slog.Logger
+}
+
+func NewSubmitter(logger *slog.Logger, client gen.MetricsClient) *Submitter {
+	return &Submitter{
+		c:      make(chan *gen.Result, 1),
+		done:   make(chan struct{}),
+		client: client,
+		logger: logger,
+	}
+}
+
+func (s *Submitter) Run(ctx context.Context) {
+	defer func() { s.done <- struct{}{} }()
+	if s.client == nil {
+		for range s.c {
```

Comment:
Why do we need to clean it?

Reply:
This whole block is a leftover from when submitter was not nil, but client code.
Removed, as this is now dead code.